pre-docs/directory_structure.md

```diff
@@ -0,0 +1,83 @@
+# Directory structures for task and plan development
+
+Bolt looks for tasks and plans following the structure of a [Puppet module](https://puppet.com/docs/puppet/latest/modules_fundamentals.html). The [Puppet Development Kit (PDK)](https://puppet.com/download-puppet-development-kit) can be used to create Puppet modules and add tasks to it.
+
+A typical module for use with Bolt may contain:
+```
+
+├── data/
+├── files/
+├── hiera.yaml
+├── lib/
+├── manifests/
+├── metadata.json
+├── plans/
+└── tasks/
+```
+
+`data/`
+    Contains Hiera data that can be used when applying a manifest block.
+`files/`
+    Contains static files that can be loaded by a plan or required as a dependency of a task. Prefer putting non-Ruby libraries used by a task here.
+`hiera.yaml`
+    The Hiera configuration for this module. See the [Hiera docs](https://puppet.com/docs/puppet/latest/hiera.html).
+`lib/`
+    Typically Ruby code such as custom Puppet functions that can be called by a plan or library code used by a task.
+`manifests/`
+    Classes and other Puppet code usable when applying a manifest block.
+`metadata.json`
+    Typical metadata for a Puppet module describing version, OS compatibility, and other module dependencies.
+`plans/`
+    Contains plans, which must end in the `.pp` extension.
+`tasks/`
+    Contains tasks and their metadata.
+
+## Developing a module
```

Comment:
@MikaelSmith This section seems a little out of place in the context of directory structures. As written, it's half about module structure, e.g., the discussion of modulepaths, and half about module-writing best practices, e.g., include automated testing, publish to a repo or the Forge. 

Would you be open to either: 1) splitting up these two topics (though I'm not sure where the module best practices would live?), or 2) tweaking this content to focus more on directory structure? I think #2 could be achieved by adding a recommendation about where to store automated tests.

Reply:
I think of these sections as three ways you might develop tasks/plans: a standalone module to publish somewhere, a module that's part of a control repository, or a module that's part of a project you want to deploy.

The testing references should be minimal, I'd be ok moving them somewhere else. We just hadn't written anything, so I included some minimally useful material.